### PR TITLE
Ussd Registration

### PR DESCRIPTION
**Pull request by gsvr**
Training ID -> Name -> Persons ID (using MCZA style) -> Gender

If user has registered before we just need to get the Training ID for their new session.
`Welcome back Mike. 
1. Register attendance
2. I am not Mike`

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review. Data posts to be handled in #6, proper state_help in #7.

File: src/registration.js [hunk 1]
```diff
@@ -35,30 +36,90 @@ go.app = function() {
...
+                        });
+                }
+            });
+        });
+
+        self.states.add('state_training_code', function(name) {
+            return new FreeText(name, {
+                question: $("What is your training session code?"),
+                next: function(choice) {
+                    // TODO: validate entered clinic code
```
Comment (imsickofmaps) [inline src/registration.js]: You mean `training code`?

File: src/registration.js [hunk 2]
```diff
@@ -66,13 +127,211 @@ go.app = function() {
...
+                            };
+                        });
+                }
+            });
+        });
+
+        self.states.add('state_birth_year', function(name, opts) {
+            var error = $('There was an error in your entry. Please carefully enter your ' +
+                          'year of birth again (for example: 2001)');
+            var question = $('Please enter the year that the pregnant mother was born' +
```
Comment (imsickofmaps) [inline src/registration.js]: O_o

File: src/registration.js [hunk 3]
```diff
@@ -35,30 +36,90 @@ go.app = function() {
...
+                }
+            });
+        });
+
+        self.states.add('state_training_code', function(name) {
+            return new FreeText(name, {
+                question: $("What is your training session code?"),
+                next: function(choice) {
+                    // TODO: validate entered clinic code
+                    return go.utils
```
Comment (imsickofmaps) [inline src/registration.js:115]: Don't you want to do this in an interstitial?

Comment (gsvr) [inline src/registration.js:115]: No, are you going to force me to?

**Comment by gsvr** [comment]
@imsickofmaps  :+1:?

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**